### PR TITLE
yang model table DEVICE_NEIGHBOR_METADATA creation

### LINKED ISSUE #10545: [yang] Need Yang for DEVICE_NEIGHBOR_METADATA table
#### Description

We don't have Yang model for DEVICE_NEIGHBOR_METADATA table.

#### Steps to reproduce the issue:
1. Run 'config apply-patch' command
2. We can get below info:
sonic_yang(6):Note: Below table(s) have no YANG models: CONSOLE_SWITCH, DEVICE_NEIGHBOR_METADATA, DHCP_SERVER, KDUMP, RESTAPI, SNMP, SNMP_COMMUNITY, TELEMETRY

#### Describe the results you received:
sonic_yang reports no Yang models.

#### Describe the results you expected:
sonic_yang reports no Yang models.

#### Output of `show version`:

```
(paste your output here)
```
#### Output of `show techsupport`:

```
(paste your output here or download and attach the file here )
```

### PR DESCRIPTION
**Pull request by jcaiMR**
#### Why I did it
Add support for DEVICE_NEIGHBOR_METADATA yang model table

#### How I did it
Create new yang files for DEVICE_NEIGHBOR_METADATA , add UT.

#### How to verify it
Run UT for sonic-yang-models

#### Which release branch to backport (provide reason below if selected)
- [x] 202205

#### A picture of a cute animal (not mandatory but encouraged)

fixes #10545

### CODE REVIEW

File: src/sonic-yang-models/yang-models/sonic-device_neighbor_metadata.yang [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+                    type string {
+                        length 1..255;
+                        pattern "ToRRouter|LeafRouter|SpineChassisFrontendRouter|ChassisBackendRouter|ASIC|MgmtToRRouter|SpineRouter|BackEndToRRouter|BackEndLeafRouter|EPMS|MgmtTsToR|BmcMgmtToRRouter|not-provisioned";
+                    }
+                }
+            }
+            /* end of list DEVICE_NEIGHBOR_METADATA_LIST */
+        }
+        /* end of container DEVICE_NEIGHBOR_METADATA */
+	}
```
Comment (qiluo-msft) [inline src/sonic-yang-models/yang-models/sonic-device_neighbor_metadata.yang:64]: Wrong indentation level?

Comment (jcaiMR) [reply src/sonic-yang-models/yang-models/sonic-device_neighbor_metadata.yang]: Thanks for comments, fixed in latest code.

File: src/sonic-yang-models/yang-models/sonic-device_neighbor_metadata.yang [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+                    type inet:ipv6-prefix;
+                }
+
+                leaf hwsku {
+                    type stypes:hwsku;
+                }
+
+                leaf type {
+                    type string {
+                        length 1..255;
```
Comment (ganglyu) [inline src/sonic-yang-models/yang-models/sonic-device_neighbor_metadata.yang:64]: with fixed pattern, no need for length

**Comment by ganglyu** [comment]
Unit tests for sonic-config-engine failed.

File: src/sonic-yang-models/yang-models/sonic-device_neighbor_metadata.yang (hunk 2 above)
Comment (jcaiMR) [reply src/sonic-yang-models/yang-models/sonic-device_neighbor_metadata.yang:64]: Thanks, will fix them in next diff. For UT failed, currently the code has some mgmt_addr type issue, which may have both CIDR and Non-CIDR format in config db, I am trying to address them.

**Comment by ganglyu** [comment]
why do you update submodule for frr?

**Comment by jcaiMR** [comment]
> why do you update submodule for frr?

Thanks for point this out, it's a miss op by commit, I will move it our.

File: src/sonic-yang-models/yang-models/sonic-device_neighbor_metadata.yang [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+
+        container DEVICE_NEIGHBOR_METADATA {
+
+            description "DEVICE_NEIGHBOR_METADATA part of config_db.json";
+
+            list DEVICE_NEIGHBOR_METADATA_LIST {
+
+                key "name";
+
+                leaf name {
```
Comment (ganglyu) [inline src/sonic-yang-models/yang-models/sonic-device_neighbor_metadata.yang:36]: Please add description for leaf.

Comment (jcaiMR) [reply src/sonic-yang-models/yang-models/sonic-device_neighbor_metadata.yang:36]: sure, will apply in the next diff

**Comment by jcaiMR** [comment]
Build failure because of https://github.com/sonic-net/sonic-buildimage/pull/11848. Owner is working on a quick fix.

**Comment by jcaiMR** [comment]
Latest build failure introduced by https://github.com/sonic-net/sonic-buildimage/pull/  11848, owner is working on fix.

**Comment by zhangyanzhao** [comment]
Can we finish the review and merge? Thanks.

**Review by ganglyu — APPROVED** [review]

**Comment by liushilongbuaa** [comment]
@yxieca , the original repo has some issue. Here is the response from github:
I have been updated that the affected pull request did not run because the forked repository was duplicated on the 29th of March 2023 and the first one was marked as inactive.

The pull request was from the first inactive repository, so the pull request points to a repository ID that doesn't exist, because it somehow was recreated.

As you mentioned you won't experience this issue from other forked repositories, but if you have other PRs from the same repository from before March 29, they might run into the same issue. Otherwise other pull requests from the repository should run as expected.

**Comment by liushilongbuaa** [comment]
@StormLiangMS , @yxieca , for this PR, auto cherry pick flow didn't work.
If feature is needed please manually cherry pick

**Pull request merged**